crash with message: "no TypeBounds allowed"

dotc crashes when comping the following code. This testcase also crashes the latest scalac.

$ cat abc.scala

```
trait T extends Array {
  def t1(as: String*): Array[String] = { varargs1(as: _*) }
  def t2(as: String*): Array[String] = { super.varargs1(as: _*) }
}
class C extends Base_1 {
  def c1(as: String*): Array[String] = { varargs1(as: _*) }
  def c2(as: String*): Array[String] = { super.varargs1(as: _*) }
}
object Test extends App {
  val t = new T {}
  println(t.t1("a", "b").mkString(","))
  println(t.t2("a", "b").mkString(","))
  val c = new C {}
  println(c.c1("a", "b").mkString(","))
  println(c.c2("a", "b").mkString(","))
}
```

$ dotc abc.scala

```
-- [E006] Missing Identifier Error: abc.scala --------------------------------------------------------------------------
2 |  def t1(as: String*): Array[String] = { varargs1(as: _*) }
  |                                         ^^^^^^^^
  |                                         not found: varargs1

longer explanation available when compiling with `-explain`
-- [E008] Member Not Found Error: abc.scala ----------------------------------------------------------------------------
3 |  def t2(as: String*): Array[String] = { super.varargs1(as: _*) }
  |                                         ^^^^^^^^^^^^^^
  |                                         value `varargs1` is not a member of Array(T.super)
-- [E006] Missing Identifier Error: abc.scala --------------------------------------------------------------------------
5 |class C extends Base_1 {
  |                ^^^^^^
  |                not found: type Base_1

longer explanation available when compiling with `-explain`
-- [E006] Missing Identifier Error: abc.scala --------------------------------------------------------------------------
6 |  def c1(as: String*): Array[String] = { varargs1(as: _*) }
  |                                         ^^^^^^^^
  |                                         not found: varargs1

longer explanation available when compiling with `-explain`
-- [E008] Member Not Found Error: abc.scala ----------------------------------------------------------------------------
7 |  def c2(as: String*): Array[String] = { super.varargs1(as: _*) }
  |                                         ^^^^^^^^^^^^^^
  |                                         value `varargs1` is not a member of Object(C.super)

exception occurred while typechecking abc.scala

exception occurred while compiling abc.scala
Exception in thread "main" java.lang.AssertionError: no TypeBounds allowed
    at dotty.tools.dotc.core.TypeApplications$$anonfun$1.apply(TypeApplications.scala:25)
    at dotty.tools.dotc.core.TypeApplications$$anonfun$1.apply(TypeApplications.scala:24)
    at scala.collection.immutable.List.loop$1(List.scala:173)
    at scala.collection.immutable.List.mapConserve(List.scala:189)
    at dotty.tools.dotc.core.TypeApplications$.argTypes$extension(TypeApplications.scala:643)
    at dotty.tools.dotc.ast.untpd$.New(untpd.scala:286)
    at dotty.tools.dotc.typer.Typer.ensureConstrCall(Typer.scala:1320)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedClassDef$1.apply(Typer.scala:1243)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedClassDef$1.apply(Typer.scala:1194)
    at dotty.tools.dotc.util.Stats$.track(Stats.scala:35)
    at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:1194)
    at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:1430)
    at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:1478)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1490)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1488)
    at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:132)
    at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:57)
    at dotty.tools.dotc.typer.Typer.typed(Typer.scala:1488)
    at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:1512)
    at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:1528)
    at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:564)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedBlock$1.apply(Typer.scala:567)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedBlock$1.apply(Typer.scala:566)
    at dotty.tools.dotc.util.Stats$.track(Stats.scala:35)
    at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:566)
    at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:1446)
    at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:1480)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1490)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1488)
    at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:132)
    at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:57)
    at dotty.tools.dotc.typer.Typer.typed(Typer.scala:1488)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedNew$1.apply(Typer.scala:438)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedNew$1.apply(Typer.scala:433)
    at dotty.tools.dotc.util.Stats$.track(Stats.scala:35)
    at dotty.tools.dotc.typer.Typer.typedNew(Typer.scala:432)
    at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:1442)
    at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:1480)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1490)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1488)
    at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:132)
    at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:57)
    at dotty.tools.dotc.typer.Typer.typed(Typer.scala:1488)
    at dotty.tools.dotc.typer.Namer.typedAheadImpl(Namer.scala:778)
    at dotty.tools.dotc.typer.Namer.typedAheadExpr(Namer.scala:788)
    at dotty.tools.dotc.typer.Namer.rhsType$1(Namer.scala:909)
    at dotty.tools.dotc.typer.Namer.cookedRhsType$1(Namer.scala:910)
    at dotty.tools.dotc.typer.Namer.lhsType$lzycompute$1(Namer.scala:911)
    at dotty.tools.dotc.typer.Namer.lhsType$1(Namer.scala:911)
    at dotty.tools.dotc.typer.Namer.inferredType$1(Namer.scala:923)
    at dotty.tools.dotc.typer.Namer.valOrDefDefSig(Namer.scala:931)
    at dotty.tools.dotc.typer.Namer$Completer.typeSig(Namer.scala:535)
    at dotty.tools.dotc.typer.Namer$Completer.completeInCreationContext(Namer.scala:602)
    at dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:561)
    at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:210)
    at dotty.tools.dotc.core.SymDenotations$SymDenotation.info(SymDenotations.scala:188)
    at dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:327)
    at dotty.tools.dotc.typer.Typer.retrieveSym(Typer.scala:1392)
    at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:1417)
    at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:1478)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1490)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1488)
    at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:132)
    at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:57)
    at dotty.tools.dotc.typer.Typer.typed(Typer.scala:1488)
    at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:1512)
    at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:1528)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedClassDef$1.apply(Typer.scala:1246)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedClassDef$1.apply(Typer.scala:1194)
    at dotty.tools.dotc.util.Stats$.track(Stats.scala:35)
    at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:1194)
    at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:1430)
    at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:1478)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1490)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1488)
    at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:132)
    at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:57)
    at dotty.tools.dotc.typer.Typer.typed(Typer.scala:1488)
    at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:1512)
    at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:1528)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedPackageDef$1.apply(Typer.scala:1346)
    at dotty.tools.dotc.typer.Typer$$anonfun$typedPackageDef$1.apply(Typer.scala:1337)
    at dotty.tools.dotc.util.Stats$.track(Stats.scala:35)
    at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:1337)
    at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:1468)
    at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:1480)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1490)
    at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1488)
    at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:132)
    at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:57)
    at dotty.tools.dotc.typer.Typer.typed(Typer.scala:1488)
    at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:1540)
    at dotty.tools.dotc.typer.FrontEnd$$anonfun$typeCheck$1.apply$mcV$sp(FrontEnd.scala:50)
    at dotty.tools.dotc.typer.FrontEnd.monitor(FrontEnd.scala:24)
    at dotty.tools.dotc.typer.FrontEnd.typeCheck(FrontEnd.scala:48)
    at dotty.tools.dotc.typer.FrontEnd$$anonfun$runOn$3.apply(FrontEnd.scala:72)
    at dotty.tools.dotc.typer.FrontEnd$$anonfun$runOn$3.apply(FrontEnd.scala:72)
    at scala.collection.immutable.List.foreach(List.scala:381)
    at dotty.tools.dotc.typer.FrontEnd.runOn(FrontEnd.scala:72)
    at dotty.tools.dotc.Run$$anonfun$compileUnits$1$$anonfun$apply$mcV$sp$1.apply(Run.scala:75)
    at dotty.tools.dotc.Run$$anonfun$compileUnits$1$$anonfun$apply$mcV$sp$1.apply(Run.scala:72)
    at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
    at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
    at dotty.tools.dotc.Run$$anonfun$compileUnits$1.apply$mcV$sp(Run.scala:72)
    at dotty.tools.dotc.Run$$anonfun$compileUnits$1.apply(Run.scala:66)
    at dotty.tools.dotc.Run$$anonfun$compileUnits$1.apply(Run.scala:66)
    at dotty.tools.dotc.util.Stats$.monitorHeartBeat(Stats.scala:76)
    at dotty.tools.dotc.Run.compileUnits(Run.scala:66)
    at dotty.tools.dotc.Run.compileSources(Run.scala:63)
    at dotty.tools.dotc.Run.compile(Run.scala:47)
    at dotty.tools.dotc.Driver.doCompile(Driver.scala:26)
    at dotty.tools.dotc.Driver.process(Driver.scala:124)
    at dotty.tools.dotc.Driver.process(Driver.scala:93)
    at dotty.tools.dotc.Driver.process(Driver.scala:105)
    at dotty.tools.dotc.Driver.main(Driver.scala:132)
    at dotty.tools.dotc.Main.main(Main.scala)
```

--- a/src/dotty/tools/dotc/ast/untpd.scala
+++ b/src/dotty/tools/dotc/ast/untpd.scala
@@ -282,7 +282,7 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
       case TypedSplice(AppliedTypeTree(tycon, targs)) =>
         (TypedSplice(tycon), targs map (TypedSplice(_)))
       case TypedSplice(tpt1: Tree) =>
-        val argTypes = tpt1.tpe.argTypes
+        val argTypes = tpt1.tpe.argTypesLo
         val tycon = tpt1.tpe.withoutArgs(argTypes)
         def wrap(tpe: Type) = TypeTree(tpe) withPos tpt.pos
         (wrap(tycon), argTypes map wrap)

--- a/src/dotty/tools/dotc/typer/Typer.scala
+++ b/src/dotty/tools/dotc/typer/Typer.scala
@@ -241,7 +241,7 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
 
       /** Would import of kind `prec` be not shadowed by a nested higher-precedence definition? */
       def isPossibleImport(prec: Int)(implicit ctx: Context) =
-        !noImports && 
+        !noImports &&
         (prevPrec < prec || prevPrec == prec && (prevCtx.scope eq ctx.scope))
 
       @tailrec def loop(implicit ctx: Context): Type = {
@@ -446,6 +446,14 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
           case _ =>
         }
         checkClassType(tpt1.tpe, tpt1.pos, traitReq = false, stablePrefixReq = true)
+
+        tpt1 match {
+          case AppliedTypeTree(_, targs) =>
+            for (targ @ TypeBoundsTree(_, _) <- targs)
+              ctx.error("type argument must be fully defined", targ.pos)
+          case _ =>
+        }
+
         assignType(cpy.New(tree)(tpt1), tpt1)
         // todo in a later phase: checkInstantiatable(cls, tpt1.pos)
     }

--- a/tests/neg/i1643.scala
+++ b/tests/neg/i1643.scala
@@ -1,0 +1,19 @@
+trait T extends Array {
+  def t1(as: String*): Array[String] = { varargs1(as: _*) } // error
+  def t2(as: String*): Array[String] = { super.varargs1(as: _*) } // error
+}
+class C extends Base_1 { // error
+  def c1(as: String*): Array[String] = { varargs1(as: _*) } // error
+  def c2(as: String*): Array[String] = { super.varargs1(as: _*) } // error
+}
+object Test extends App {
+  val t = new T {}  // error
+  println(t.t1("a", "b").mkString(","))
+  println(t.t2("a", "b").mkString(","))
+  val c = new C {}
+  println(c.c1("a", "b").mkString(","))
+  println(c.c2("a", "b").mkString(","))
+
+  class CC[T]
+  val x = new CC[_]  // error
+}

